### PR TITLE
[@xstate/store] Compatibility with `@xstate/react`'s `useSelector(…)` hook

### PR DESCRIPTION
**Pull request by davidkpiano**

### CODE REVIEW

File: packages/xstate-react/src/useSelector.ts [hunk 1]
```diff
@@ -10,10 +10,15 @@ function defaultCompare<T>(a: T, b: T) {
   return a === b;
 }
 
-export function useSelector<TActor extends AnyActorRef | undefined, T>(
+export function useSelector<
+  TActor extends Pick<AnyActorRef, 'subscribe' | 'getSnapshot'> | undefined,
```
Comment (Andarist) [inline packages/xstate-react/src/useSelector.ts:14]: why is the previous version not compatible?

Comment (davidkpiano) [reply packages/xstate-react/src/useSelector.ts:14]: Because `Store` is not fully compatible with `AnyActorRef`

Comment (Andarist) [reply packages/xstate-react/src/useSelector.ts:14]: Hm, ok - lots of things are missing from this. The solution is OK 👍 I would do the same in all of the other integration packages though. `@xstate/react` shouldn't be the only one that becomes compatible with this

File: .changeset/tall-bobcats-trade.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+---
+'@xstate/react': patch
+'@xstate/store': patch
```
Comment (Andarist) [inline .changeset/tall-bobcats-trade.md:3]: To be fair - this change doesn't really affect `@xstate/store` so it feels off to me that it's going to be mentioned in its CHANGELOG cause of this changeset. The change itself is solely in `@xstate/react`

File: packages/xstate-react/src/useSelector.ts (hunk 1 above)
Comment (davidkpiano) [reply packages/xstate-react/src/useSelector.ts:14]: Agreed

File: .changeset/tall-bobcats-trade.md (hunk 2 above)
Comment (Andarist) [reply .changeset/tall-bobcats-trade.md:3]:
```suggestion
```

**Review by Andarist — APPROVED** [review]

**Pull request merged**